GeometryTools/Size.swift

```diff
@@ -25,9 +25,20 @@ public struct Size {
         self.height = height
     }
 
+    /// - Returns: a `Size` with both dimensions scaled by the same value.
     public func scaled(by value: Double) -> Size {
         return Size(width: width * value, height: height * value)
     }
+
+    /// - Returns: a `Size` scaled by the given dimensions.
+    public func scaled(byWidth widthScale: Double = 1, byHeight heightScale: Double = 1) -> Size {
+        return Size(width: width * widthScale, height: height * heightScale)
```

Comment:
Then, here, maybe you could just call the higher-level one below it:

```Swift
return scaled(by: Size(width: widthScale, height: heightScale))
```

Which gets rid of one repetition of calculation. Minor, though.